client/modules/workspace/src/AppsSubmissionForm/AppsSubmissionForm.tsx

```diff
@@ -419,14 +419,15 @@ export const AppsSubmissionForm: React.FC = () => {
             [sParameterSet]: Object.entries(sParameterValue)
               .map(([k, v]) => {
                 if (!v) return;
+                const field = parameterSet.fields?.[sParameterSet]?.[k];
```

Comment:
parameterSet.fields gives list of all parameterSet types. sParameterSet is the key to those types. For example: parameterSet.fields.envVariables gives all env vars.